### PR TITLE
Fix Account Purger's CloudWatch alarm

### PR DESCRIPTION
**Pull request by hacodeorg**
[FND-1360](https://codedotorg.atlassian.net/browse/FND-1360)
Fix Account Purger's CloudWatch alarm to use the correct value of `ManualReviewQueueDepth`.
To avoid confusion, Account Purger will start reporting both number of accounts in the review queue (`ReviewQueueDepth`) and number of accounts require manual reviews (`ManualReviewQueueDepth`). 

## Context
From 8/24/2020 up to 1/10/2021, the Account Purger's CloudWatch alarm consistently showed almost 7K accounts **needed manual reviews**, much higher than the threshold value of 30.
Screen Shot 2021-01-17 at 10 42 27 AM

However, the Account Purger's Slack report in #user-accounts channel showed <30 accounts required manual reviews. 
Screen Shot 2021-01-17 at 11 12 21 AM

The reason was the CloudWatch alarm actually shows the **total number of accounts queued** in the `QueuedAccountPurge`. It includes accounts that we failed to purge for any reasons, including auto-retryable ones. Only a small portion of them required manual reviews. 

## Links:
* Purged Accounts [CloudWatch dashboard](https://console.aws.amazon.com/cloudwatch/home?region=us-east-1#dashboards:name=Purged-Accounts).
* Queued Account Purge [CloudWatch alarm](https://console.aws.amazon.com/cloudwatch/home?region=us-east-1#alarmsV2:alarm/Queued+Account+Purge?~(search~'account)) --> [metric](https://console.aws.amazon.com/cloudwatch/home?region=us-east-1#metricsV2:graph=~(region~'us-east-1~metrics~(~(~'DeletedAccountPurger~'ManualReviewQueueDepth~'Environment~'production~(stat~'Average)))~view~'timeSeries~stacked~false~start~'-P3M~end~'P0D~period~86400~annotations~(horizontal~(~(label~'ManualReviewQueueDepth*20*3e*3d*2030*20for*201*20datapoints*20within*201*20day~value~30)))~title~'Queued*20Account*20Purge))
* Example of an [AccountPurger message](https://codedotorg.slack.com/archives/C6B838SB0/p1610870636000400) in the #cron-daily channel.

## Testing story
* Unit test: `bundle exec rails test test/lib/expired_deleted_account_purger_test.rb`
* Test in Rails console:
  ```
  edap = ExpiredDeletedAccountPurger.new
  edap.send(:report_results)
  edap.log.string
  # Verify that the log string includes ReviewQueueDepth and ManualReviewQueueDepth
  ```

### CODE REVIEW

File: dashboard/test/lib/expired_deleted_account_purger_test.rb [hunk 1]
```diff
@@ -481,7 +493,8 @@ def teardown
       Done purging user_id #{students[5].id}
       AccountsPurged: 6
       AccountsQueued: 0
-      ManualReviewQueueDepth: #{QueuedAccountPurge.all.count}
+      ReviewQueueDepth: #{QueuedAccountPurge.count}
```
Comment (daynew) [inline dashboard/test/lib/expired_deleted_account_purger_test.rb:496]: This used to have `.all`. Why don't we want it anymore?

File: dashboard/lib/expired_deleted_account_purger.rb [hunk 2]
```diff
@@ -156,14 +157,16 @@ def report_results
...
-  def build_metrics(review_queue_depth)
+  def build_metrics(review_queue_depth, manual_reviews_needed)
     {
       # Number of accounts purged during this run
       AccountsPurged: @num_accounts_purged,
       # Number of accounts queued for manual review during this run
       AccountsQueued: @num_accounts_queued,
       # Depth of review queue after this run (may include auto-retryable entries)
-      ManualReviewQueueDepth: review_queue_depth,
+      ReviewQueueDepth: review_queue_depth,
```
Comment (daynew) [inline dashboard/lib/expired_deleted_account_purger.rb:167]: Do you know why `QueuedAccountPurge.count` was so high and not decreasing? If these are accounts which should be auto cleaned up, why was the queue never decreasing in size?

**Review by daynew — APPROVED** [review]
Nice find that we are logging the wrong metric.

I am wondering why the previous number was very high and not decreasing though...

File: dashboard/test/lib/expired_deleted_account_purger_test.rb (hunk 1 above)
Comment (hacodeorg) [reply dashboard/test/lib/expired_deleted_account_purger_test.rb:496]: We just want to count number of records in the `QueuedAccountPurge` model, using `.all` was redundant.

File: dashboard/lib/expired_deleted_account_purger.rb (hunk 2 above)
Comment (hacodeorg) [reply dashboard/lib/expired_deleted_account_purger.rb:167]: I don't really know what happened. The queue started to shrink from Jan 10th, maybe related to something you did here?
Screen Shot 2021-01-22 at 10 32 59 AM
Screen Shot 2021-01-22 at 10 33 13 AM

**Pull request merged**